src/Sushi.Mediakiwi.API/Controllers/Authentication.cs

```diff
@@ -0,0 +1,163 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Threading.Tasks;
+using Sushi.Mediakiwi.API.Services;
+using Sushi.Mediakiwi.API.Transport.Requests;
+using Sushi.Mediakiwi.API.Transport.Responses;
+using System;
+
+namespace Sushi.Mediakiwi.API.Controllers
+{
+    [ApiController]
+    [MediakiwiApiAuthorize]
+    [Route(Common.MK_CONTROLLERS_PREFIX + "authentication")]
+    [ResponseCache(NoStore = true, Location = ResponseCacheLocation.None)]
+    public class Authentication : ControllerBase
+    {
+        private readonly IUserService userService;
+
+        public Authentication(IUserService _service)
+        {
+            userService = _service;
+        }
+
+        /// <summary>
+        /// Logs a user in based on their Email / Password combination. 
+        /// This method will also set a cookie 'MKAPI' which contains the JWT token.
+        /// The default Expiration time is 2 hours
+        /// </summary>
+        /// <param name="request">The request containing the needed credentials</param>
+        /// <returns></returns>
+        /// <response code="200">The user is succesfully authenticated</response>
+        /// <response code="400">Some needed information is missing from the request</response>
+        /// <response code="401">The user is not succesfully authenticated</response>
+        [AllowAnonymous]
+        [Produces("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [HttpPost("Login")]
+        public async Task<ActionResult<LoginResponse>> Login([FromBody]LoginRequest request)
+        {
+            if (request == null || userService == null || string.IsNullOrWhiteSpace(request.ApiKey) || string.IsNullOrWhiteSpace(request.EmailAddress) || string.IsNullOrWhiteSpace(request.Password))
+            {
+                return BadRequest();
+            }
+
+            // Compare API key with environment
+            var env = await Data.Environment.SelectOneAsync().ConfigureAwait(false);
+            if (env.ApiKey.Equals(request.ApiKey, StringComparison.InvariantCulture) == false)
+            {
+                return Unauthorized();
+            }
+
+            // Check for user existence and create jwtToken if valid.
+            LoginResponse result = await userService.LoginAsync(request).ConfigureAwait(false);
+
+            // When successful, return filled response object
+            if (result.StatusCode == System.Net.HttpStatusCode.OK)
+            {
+                var cookieOptions = new CookieOptions
```

Comment:
use cookie policy middleware to set cookie options for all cookies:
https://docs.microsoft.com/en-us/aspnet/core/security/authentication/cookie?view=aspnetcore-6.0#cookie-policy-middleware